### ISSUE TITLE
Steam Offline Issue? Please Check Here First! (Solution)

### ISSUE THREAD

**Issue by ABaumher** [FriendsOfGalaxy/galaxy-integration-steam#178]
Hello all, 

I, like many of you, have noticed an issue with the Steam Plugin in GOG. I have posted on #159 previously, but am posting here for visibility. I am obusing the Nature of a bug report, but considering it has been well-established the maintainer here is no longer active, i don't believe i will have any issues with posting here. 

With the help of a great number of other developers, I have taken over the maintenance of this plugin, albeit at a different repo and in without direct integration to GOG like this plugin originally had. It is my hope that eventually the organization i am part of will be able to take over with GOG to directly provide them like this maintainer originally did, but I cannot promise this at this time. 

I would like to provide what is currently a viable solution. I am part of an small collective of individuals who maintain several plugins that used to be maintained here. We are active at https://github.com/GOG-Nebula/ and, crucially, we have a patched plugin for Steam at https://github.com/GOG-Nebula/galaxy-integration-steam/ 
I also maintain a development branch on my own repo with changes from other developers attempting to improve the overall product. 1.0.7 was a heavily patched version of 1.0.4 on this repo, with a great deal of changes under the hood, but it's still heavily dependent on the code here, which had some issues to begin with. I am working on a 2.0 release but it has taken a backseat to other, more pressing issues in my life, which i will not apologize for.

Tl;Dr: Releases here:
[V 1.0.7 (Stable)](https://github.com/GOG-Nebula/galaxy-integration-steam/releases/tag/v1.0.7) 
[V 1.2.0 (Beta)](https://github.com/ABaumher/galaxy-integration-steam/releases/tag/v1.2.0beta)

Install instructions are on the respective repos. If you choose the version with batch scripts, they will work largely out of the box: unzip the release, run the command or batch file, and you're good to go. Please note that Mac users on the latest OS may have to manually allow the script to run; MacOS is extremely locked down and we did the best we could without paying Apple exorbidant amounts of money. 

As with most GOG plugins, our code is completely open-source; you can review it for yourself if you are wary trusting a stranger on the internet (which is totally fair). 

This project was community-driven, and includes contributions from the following:
 @TheSentry, @LauraRozier, @slo, @don-de-marco, @MisterZeus, @wwMRd, and possibly others (apologies)

If you have any issues, feel free to raise issues on the respective repos. Also, to keep this as clean as possible, please try to limit responses here so more people can see this. I'm still subscribed to #159, so feel free to comment there instead with appreciation or suggestions.

**Comment by ABaumher** [comment]
**UPDATE**

There has been an issue where your credentials expire. We knew this would happen, but could not trick steam into giving us this error before now. Some users were able to point this out for us, and we have put in a quick fix. Those of you getting automatic updates via Nebula will have to wait a little, these changes, no matter how small, must be reviewed first. For the rest of you, you can get them off my repo.

[1.0.8 (Stable)](https://github.com/ABaumher/galaxy-integration-steam/releases/tag/v1.0.8)
[1.2.1 (Beta)](https://github.com/ABaumher/galaxy-integration-steam/releases/tag/v1.2.1Beta)

Please note that if you want to wait for the official Nebula release and have these issues, you can simply log out and log back in again by disconnecting the plugin frm the Plugins menu, then reconnecting it. This will buy you another seven months for the automatic update to kick in.

**Comment by mortenanakin** [comment]
Beta and stable gave me the "El plugin se ha colgado" error, and retry option doesn't works :/

**Comment by mortenanakin** [comment]
![image](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/assets/52761884/5696b183-3dcf-4116-aaf9-86076d9489f4)

**Comment by ABaumher** [comment]
I forgot a colon, give me a minute to fix

**Comment by mortenanakin** [comment]
Thanks, no rush! :)

**Comment by LeyendaV** [comment]
The manual installation instructions on the repo literally only says "this is too hard to install manually, use the version with an installer instead".
Can we have actual instructions?

**Comment by ABaumher** [comment]
@LeyendaV nebula or my repo?

**Comment by Danyelalejandro** [comment]
Hello all, with new version I am able to connect but it doesn't import nothing from library, achievements or play time

**Comment by otede** [comment]
So what's the current solution? It's March 2024. The "fixed build" linked in the OP is from summer 2023 and the steam integration by https://github.com/FriendsOfGalaxy (THIS repo) is from Nov 2023.

**Comment by ABaumher** [comment]
Friends of Galaxy is inactive and has been for a while. In essence, _this_ repo is dead. The links posted are the latest community-driven patch, from roughly Christmas of this past year. It's a few comments down. Last i checked, the plugin works with these patches (excepting extremely large libraries, which never worked to begin with. A fix is planned but has no timetable). 

An aside, if your metric for whether or not something works is how recently it had changes, you're understanding of how development works is somewhat flawed. You don't do active development on code that works, after all, you do it on buggy code. Once something is stable, there's little reason to change it, except for bugfixes (which caused the December update). That's not to say the current implementation is flawless, but it works well for most users. When there's code worth updating, we'll do so.

**Comment by ImJezze** [comment]
Unfortunately, it is still not working for me. The connection status is Offline and clicking in retry does seem to work. I checked the log file of the plugin and it shows several errors. Maybe you can have a look. 

[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/15365948/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)

**Comment by sean8102** [comment]
You're plugin worked for me no problem. Hope GoG dose decide to use your plugin instead inside of Galaxy because the "FriendsOfGalaxy" one seems to be flat out broken. I had no issue syncing my Steam account to Playnite.

But as I said after installing your Steam integration for Galaxy (thank you for including a simple installer batch file) it worked perfectly first time I tried it.

**Comment by bika95** [comment]
> ![image](https://private-user-images.githubusercontent.com/52761884/292918156-5696b183-3dcf-4116-aaf9-86076d9489f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIzNjA0ODEsIm5iZiI6MTczMjM2MDE4MSwicGF0aCI6Ii81Mjc2MTg4NC8yOTI5MTgxNTYtNTY5NmIxODMtM2RjZi00MTE2LWFhZjktODYwNzZkOTQ4OWY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIzVDExMDk0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhY2EzZWQyZGVhOWRlMDUwODU2N2NkNWZkOTZmMjY5M2NkMThiOTFkM2E5NjQ1NjZkZmM2MDBjZWE0MjYxNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1CNBBb370nA0Jd8iEkF9qZdXr8WCe3x2WyEnVUeWjOY)


This error happened to me, but I installed the plugin with GOG Galaxy open, I believe this was the problem. When I installed it after closing GOG Galaxy, it worked perfectly.

**Comment by dontknow-whatimdoing** [comment]
Followed instructions for 1.0.8 with installer and FINALLY got the steam integration working.

**Comment by Danyelalejandro** [comment]
> Followed instructions for 1.0.8 with installer and FINALLY got the steam integration working.

Mine doesnt. Only imports a part of the games, never imports them all.

**Comment by MisterZeus** [comment]
> > Followed instructions for 1.0.8 with installer and FINALLY got the steam integration working.
> 
> Mine doesnt. Only imports a part of the games, never imports them all.

How many Steam games in total? If you have thousands, the Python and Steam APIs groan under the load...

**Comment by Danyelalejandro** [comment]
> > > Followed instructions for 1.0.8 with installer and FINALLY got the steam integration working.
> > 
> > 
> > Mine doesnt. Only imports a part of the games, never imports them all.
> 
> How many Steam games in total? If you have thousands, the Python and Steam APIs groan under the load...

Thousands, yes
